src/js/utils/ajax.js

```diff
@@ -158,13 +160,19 @@ function _readyStateChangeHandler(options) {
         const xhr = e.currentTarget || options.xhr;
         if (xhr.readyState === 4) {
             clearTimeout(options.timeoutId);
-            if (xhr.status >= 400) {
-                _error(options, MSG_CANT_PLAY_VIDEO, (xhr.status < 600) ? xhr.status : ERROR_XHR_UNKNOWN);
+            const status = xhr.status;
+            if (status >= 400) {
+                _error(options, MSG_CANT_PLAY_VIDEO, (status < 600) ? status : ERROR_XHR_UNKNOWN);
                 return;
             }
-            if (xhr.status === 200) {
+            if (status === 200) {
                 return _ajaxComplete(options)(e);
             }
+            // regex checks that the url is relative or protocol relative
+            if (status === 0 && isFileProtocol() && !/^(?:(?:https?|file):\/\/)/.test(options.url)) {
```

Comment:
Can you explain what the regex is doing?

Reply:
@vseventer there is a comment above the regex; did you miss it or was it unclear when you read it ?

Reply:
Yeah, I didn't realize `isFileProtocol` returns the current page protocol.. Either way, can't the regex be shortened to `/^(https?|file):\/\//`, since the `RegExp.test` method returns a `boolean`?

Reply:
I agree 100%, `!/^[a-z][a-z0-9+.-]*:/.test(options.url)` is better